eth/downloader/downloader_test.go

```diff
@@ -1431,13 +1431,32 @@ func (w *whitelistFake) IsValidChain(current *types.Header, headers []*types.Hea
 }
 func (w *whitelistFake) ProcessCheckpoint(_ uint64, _ common.Hash) {}
 
-func (w *whitelistFake) GetCheckpointWhitelist() map[uint64]common.Hash {
-	return nil
+func (w *whitelistFake) GetWhitelistedCheckpoint() (bool, uint64, common.Hash) {
+	return false, 0, common.Hash{}
+}
+func (w *whitelistFake) PurgeWhitelistedCheckpoint() {}
+
+func (w *whitelistFake) ProcessMilestone(_ uint64, _ common.Hash) {}
+func (w *whitelistFake) GetWhitelistedMilestone() (bool, uint64, common.Hash) {
+	return false, 0, common.Hash{}
 }
-func (w *whitelistFake) PurgeCheckpointWhitelist() {}
+func (w *whitelistFake) PurgeWhitelistedMilestone() {}
+
 func (w *whitelistFake) GetCheckpoints(current, sidechainHeader *types.Header, sidechainCheckpoints []*types.Header) (map[uint64]*types.Header, error) {
 	return map[uint64]*types.Header{}, nil
 }
+func (w *whitelistFake) LockMutex(endBlockNum uint64) bool {
```

Comment:
I believe this methods LockMutex, UnlockMutex, UnlockSprint shall be removed in prefer to more simple solution with no mutex dancing.